### PR TITLE
No need to emit additonalscript event on public pages

### PR DESCRIPTION
**Pull request by rullzer**
There already is a separate event for this. This will make it possible
to only inject code with the logged in one on default rendered pages.

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

**Comment by nickvergessen** [comment]
Could we exclude `/login/` too? That is the two factor pages, where the user is already logged in, but not done with authentication :(
https://github.com/nextcloud/notifications/blob/master/lib/AppInfo/Application.php#L75

**Comment by rullzer** [comment]
@nickvergessen ha good point. Lets see how we can do that. Maybe with a separate interface for those pages? I'd like to avoid relying on url magic.

**Comment by nickvergessen** [comment]
> I'd like to avoid relying on url magic.

:scream: No 🔮 for you?
Yeah sounds sane

**Comment by rullzer** [comment]
Ok updated. Lets see if CI is happy

File: lib/public/AppFramework/Http/StandAloneTemplateResponse.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * A template response that does not emit the loadAdditionalScripts events.
+ *
+ * This is useful for pages that are authenticated but do not yet show the
+ * full nextcloud UI. Like the 2FA page, or the grant page in the login flow.
+ *
+ * @since 16.0.0
+ */
+class StandAloneTemplateResponse extends TemplateResponse {
```
Comment (MorrisJobke) [inline lib/public/AppFramework/Http/StandAloneTemplateResponse.php]: Isn't "standalone" one word? Then it would be `StandaloneTemplateResponse`, or not?

Comment (rullzer) [reply lib/public/AppFramework/Http/StandAloneTemplateResponse.php]: yes I guess you are right. Will fix

Comment (rullzer) [reply lib/public/AppFramework/Http/StandAloneTemplateResponse.php]: Done

**Comment by rullzer** [comment]
@nickvergessen @MorrisJobke please have another look.

File: lib/private/AppFramework/Middleware/AdditionalScriptsMiddleware.php [hunk 2]
```diff
@@ -42,6 +44,14 @@ public function __construct(EventDispatcherInterface $dispatcher, IUserSession $
 	}
 
 	public function afterController($controller, $methodName, Response $response): Response {
+		/*
+		 * There is no need to emit these signals on a public share page
+		 * There is a separate event for that already
+		 */
+		if ($controller instanceof PublicShareController || $controller instanceof StandaloneTemplateResponse) {
```
Comment (MorrisJobke) [inline lib/private/AppFramework/Middleware/AdditionalScriptsMiddleware.php]:
```suggestion
		if ($controller instanceof PublicShareController || $response instanceof StandaloneTemplateResponse) {
```

Or did I miss something?

Comment (rullzer) [reply lib/private/AppFramework/Middleware/AdditionalScriptsMiddleware.php]: It of course helps if I push the latest changes :S sorry.
Anyway fixed and with tests

File: lib/public/AppFramework/Http/StandaloneTemplateResponse.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ */
+
+namespace OCP\AppFramework\Http;
+
+/**
+ * A template response that does not emit the loadAdditionalScripts events.
+ *
+ * This is useful for pages that are authenticated but do not yet show the
+ * full nextcloud UI. Like the 2FA page, or the grant page in the login flow.
```
Comment (MorrisJobke) [inline lib/public/AppFramework/Http/StandaloneTemplateResponse.php]: Doesn't work on the normal login page.

Comment (rullzer) [reply lib/public/AppFramework/Http/StandaloneTemplateResponse.php]: The login page is not authenticated right?

Comment (rullzer) [reply lib/public/AppFramework/Http/StandaloneTemplateResponse.php]: Ah you mean that it should emit the regular loadAdditionalScripts on the login page?

Comment (MorrisJobke) [reply lib/public/AppFramework/Http/StandaloneTemplateResponse.php]: > Ah you mean that it should emit the regular loadAdditionalScripts on the login page?

Correct :)

Comment (rullzer) [reply lib/public/AppFramework/Http/StandaloneTemplateResponse.php]: fixed

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**